### PR TITLE
test(auth): use supabase cli for most - docker for rest

### PR DESCRIPTION
**Pull request by mandarini**
## Summary

This PR migrates the `auth-js` test infrastructure from a multi-instance Docker Compose setup to a single Supabase CLI instance for the majority of tests, while preserving Docker-based tests for edge cases that require specific GoTrue configurations.

## Motivation

The original `auth-js` test setup used Docker Compose to run 4 separate GoTrue instances with different configurations:

| Port | Configuration | Use Case |
|------|---------------|----------|
| 9999 | Signup enabled, autoconfirm OFF | Email verification flows, SMS tests |
| 9998 | Signup enabled, autoconfirm ON | Standard auth flows |
| 9997 | Signup disabled, autoconfirm OFF | Signup disabled tests |
| 9996 | Signup enabled, asymmetric keys | RS256 JWT verification |

We have decided to move our test suites to use the `supabase` CLI wherever possible. It provides a simpler, faster alternative that covers the majority of test scenarios with a single instance.

### New Test Structure

```
packages/core/auth-js/test/
├── lib/
│   └── clients.ts          # CLI clients only (simplified)
├── docker-tests/
│   ├── clients.ts          # Docker clients (self-contained)
│   ├── signup-disabled.test.ts
│   ├── asymmetric-jwt.test.ts
│   ├── phone-otp.test.ts
│   └── anonymous-disabled.test.ts
├── supabase/
│   └── config.toml         # Supabase CLI configuration
├── GoTrueClient.test.ts    # Main tests (CLI)
├── GoTrueApi.test.ts       # Admin API tests (CLI)
└── ...other test files
```

### Test Distribution

| Test Suite | Runner | Tests | Description |
|------------|--------|-------|-------------|
| `test:auth` | Supabase CLI | 331 | Standard auth flows |
| `test:docker` | Docker Compose | 11 | Edge cases requiring specific configs |

### Docker-Only Tests (11 total)

These tests require GoTrue configurations not possible with a single CLI instance:

1. **Signup Disabled** (1 test)
   - Requires `GOTRUE_DISABLE_SIGNUP=true`

2. **Asymmetric JWT / RS256** (3 tests)
   - Requires GoTrue configured with RS256 signing keys
   - CLI uses HS256 by default

3. **Phone OTP / SMS** (6 tests)
   - Requires Twilio SMS provider configuration
   - CLI does not include SMS provider setup

4. **Anonymous Sign-in Disabled** (1 test)
   - Requires `GOTRUE_EXTERNAL_ANONYMOUS_USERS_ENABLED=false`

### Targets/Scripts to run

| Target | Description |
|--------|-------------|
| `test:auth` | Full CLI test lifecycle (331 tests) |
| `test:infra` | Start Supabase CLI instance |
| `test:suite` | Run CLI tests only |
| `test:docker` | Full Docker test lifecycle (11 tests) |
| `test:docker:infra` | Start Docker containers |
| `test:docker:suite` | Run Docker tests only |
```

### CI Configuration

The CI workflow runs both test suites in parallel:

```yaml
jobs:
  setup-build-test-node-20:
    # Runs test:auth (CLI tests) - 331 tests
    
  test-auth-js-docker:
    # Runs test:docker (Docker tests) - 11 tests
```

## Supabase CLI Configuration

Key settings in `test/supabase/config.toml`:

```toml
[auth]
enable_anonymous_sign_ins = true

[auth.email]
enable_confirmations = false  # Autoconfirm ON

[auth.sms]
enable_signup = true
max_frequency = "1s"

[auth.sms.test_otp]
"1234567890" = "123456"

[auth.mfa.totp]
enroll_enabled = true
verify_enabled = true

[auth.mfa.phone]
enroll_enabled = true
verify_enabled = true
```

### Run CLI tests locally
```bash
npx nx test:auth auth-js
```

### Run Docker tests locally
```bash
npx nx test:docker auth-js
```

## Bug Fixes

This PR also fixes a pre-existing test bug:

- **fetchJwk tests leaking mocked fetch** - The `fetchJwk` tests were overriding `authWithSession['fetch']` with a mock but never restoring it. This caused subsequent tests (like `signInAnonymously`) to fail because they used the mocked fetch that returned JWKS data instead of auth responses. Fixed by adding `beforeEach`/`afterEach` to save and restore the original fetch.

### CODE REVIEW

File: packages/core/auth-js/test/supabase/signing_keys.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[
```
[the remark this answers is not in the document]
Comment (staaldraad) [reply packages/core/auth-js/test/supabase/signing_keys.json]: test keys, not active or used anywhere other than in these tests. Tests require valid keys

File: packages/core/auth-js/test/lib/clients.ts [hunk 2]
```diff
@@ -1,27 +1,23 @@
...
-export const SIGNUP_ENABLED_ASYMMETRIC_AUTO_CONFIRM_ON_PORT = 9996
+// Supabase CLI JWT secret
+export const GOTRUE_JWT_SECRET = 'super-secret-jwt-token-with-at-least-32-characters-long'
 
-export const GOTRUE_URL_SIGNUP_ENABLED_AUTO_CONFIRM_OFF = `http://localhost:${SIGNUP_ENABLED_AUTO_CONFIRM_OFF_PORT}`
-export const GOTRUE_URL_SIGNUP_ENABLED_AUTO_CONFIRM_ON = `http://localhost:${SIGNUP_ENABLED_AUTO_CONFIRM_ON_PORT}`
-export const GOTRUE_URL_SIGNUP_ENABLED_ASYMMETRIC_AUTO_CONFIRM_ON = `http://localhost:${SIGNUP_ENABLED_ASYMMETRIC_AUTO_CONFIRM_ON_PORT}`
-export const GOTRUE_URL_SIGNUP_DISABLED_AUTO_CONFIRM_OFF = `http://localhost:${SIGNUP_DISABLED_AUTO_CONFIRM_OFF_PORT}`
+// Supabase CLI default anon key (RS256-signed, required for API gateway)
+const SUPABASE_ANON_KEY =
```
[the remark this answers is not in the document]
Comment (staaldraad) [reply packages/core/auth-js/test/lib/clients.ts]: test keys, not active or used anywhere other than in these tests. Tests require valid keys
